com/counter_test.go

```diff
@@ -0,0 +1,32 @@
+package com
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestCounter_Add(t *testing.T) {
+	var c Counter
+
+	c.Add(42)
+	require.Equal(t, uint64(42), c.Val(), "unexpected value")
+	require.Equal(t, uint64(42), c.Total(), "unexpected total")
+
+	c.Add(23)
+	require.Equal(t, uint64(65), c.Val(), "unexpected new value")
+	require.Equal(t, uint64(65), c.Total(), "unexpected new total")
+}
+
+func TestCounter_Reset(t *testing.T) {
+	var c Counter
+
+	c.Add(42)
+	c.Reset()
```

Comment:
Why not check the return value?

```suggestion
	require.Equal(t, uint64(42), c.Reset(), "unexpected reset value")
```